internal/controllers/device_data_controller.go

```diff
@@ -151,9 +151,9 @@ func (udc *UserDevicesController) QueryDeviceErrorCodes(c *fiber.Ctx) error {
 		return err
 	}
 
-	dd, err := udc.DeviceDefSvc.GetDeviceDefinitionByID(c.Context(), ud.DeviceDefinitionID)
+	dd, err := udc.DeviceDefSvc.GetDeviceDefinitionByID(c.Context(), ud.DefinitionID.String)
```

Comment:
this looks like changing the underlying way the GRPC endpoint works as to what key it uses... Is there an accompanying PR for this change? I don't know how many places use that endpoint but I think it is a fair amount, I would be more comfortable with doing a v2 of it, transitioning searching in all repos, then moving it over.